src/bufr/BufrParser/BufrParser.cpp

```diff
@@ -7,137 +7,141 @@
 
 #include "BufrParser.h"
 
-#include <map>
 #include <ostream>
 #include <iostream>
+#include <chrono>  // NOLINT
 
 #include "eckit/exception/Exceptions.h"
 
-#if __has_include("bufr_interface.h")  // TODO(rmclaren): Remove this in future
-    #include "bufr_interface.h"
-#else
-    #include "bufr.interface.h"
-#endif
-
-#include "BufrParser/BufrCollectors/BufrCollectors.h"
-#include "BufrMnemonicSet.h"
 #include "DataContainer.h"
+#include "DataObject.h"
 #include "Exports/Export.h"
 #include "Exports/Splits/Split.h"
 
+#include "Query/QuerySet.h"
+
 
-namespace Ingester
-{
+namespace Ingester {
     BufrParser::BufrParser(const BufrDescription &description) :
-        description_(description),
-        fortranFileId_(0),
-        table1FileId_(0),
-        table2FileId_(0)
+            description_(description),
+            file_(bufr::File(description_.filepath(),
+                             description_.isWmoFormat(),
+                             description_.tablepath()))
     {
-        reset();
     }
 
-    BufrParser::BufrParser(const eckit::Configuration& conf) :
-        description_(BufrDescription(conf)),
-        fortranFileId_(0),
-        table1FileId_(0),
-        table2FileId_(0)
+    BufrParser::BufrParser(const eckit::Configuration &conf) :
+            description_(BufrDescription(conf)),
+            file_(bufr::File(description_.filepath(),
+                             description_.isWmoFormat(),
+                             description_.tablepath()))
     {
-        reset();
     }
 
     BufrParser::~BufrParser()
     {
-        closeBufrFile();
+        file_.close();
     }
 
-    std::shared_ptr <DataContainer> BufrParser::parse(const size_t maxMsgsToParse)
+    std::shared_ptr<DataContainer> BufrParser::parse(const size_t maxMsgsToParse)
     {
-        const unsigned int SubsetStringLength = 25;
+        auto startTime = std::chrono::steady_clock::now();
 
-        if (fortranFileId_ <= 10)
+        std::cout << "Start" << std::endl;
+        auto querySet = bufr::QuerySet();
+        for (const auto &var : description_.getExport().getVariables())
         {
-            throw eckit::BadValue("Fortran File ID is an invalid number (must be > 10).");
+            for (const auto &queryPair : var->getQueryList())
+            {
+                querySet.add(queryPair.name, queryPair.query);
+            }
         }
 
-        auto collectors = BufrCollectors(fortranFileId_);
-        collectors.addMnemonicSets(description_.getMnemonicSets());
+        std::cout << "Executing Queries" << std::endl;
+        const auto resultSet = file_.execute(querySet, maxMsgsToParse);
 
-        char subset[SubsetStringLength];
-        int iddate;
-
-        unsigned int messageNum = 0;
-        while (ireadmg_f(fortranFileId_, subset, &iddate, SubsetStringLength) == 0)
+        std::cout << "Building Bufr Data" << std::endl;
+        auto srcData = BufrDataMap();
```

Comment:
`auto` is  just a c++ keyword that tells the compiler it should figure out a return type for you. It's a newish feature introduced in c++ 11. It makes the code look cleaner because it avoids a lot of redundant information, for example:

`std::string myString = std::string(dataBuffer, 20)` vs `auto myString = std::string(dataBuffer, 20)`

Also it makes it easier to change things if you want to say myString to be something other than a std::string... Some template tricks also depend on this feature as you can return `auto` and let the compiler figure out the return type from a templated function for you...